engine/order_manager_test.go

```diff
@@ -1658,3 +1683,24 @@ func TestProcessFuturesPositions(t *testing.T) {
 		t.Errorf("received '%v', expected '%v'", err, nil)
 	}
 }
+
+// TestGetByDetail tests orderstore.getByDetail
+func TestGetByDetail(t *testing.T) {
+	t.Parallel()
+	m := OrdersSetup(t)
+	assert.Nil(t, m.orderStore.getByDetail(nil), "Fetching a nil order should return nil")
+	od := &order.Detail{
+		Exchange:      testExchange,
+		AssetType:     asset.Spot,
+		OrderID:       "AdmiralHarold",
+		ClientOrderID: "DuskyLeafMonkey",
+	}
+	concise := &order.Detail{
+		Exchange: od.Exchange,
+		OrderID:  od.OrderID,
+	}
+	assert.Nil(t, m.orderStore.getByDetail(od), "Fetching a non-stored order should return nil")
+	assert.Nil(t, m.orderStore.add(od), "Adding the details should not error")
+	assert.Same(t, m.orderStore.getByDetail(od), od, "Retrieve by full details should return the same pointer")
+	assert.Same(t, m.orderStore.getByDetail(concise), od, "Retrieve by other details should return the original pointer")
```

Comment:
Suggestion: I prefer not to expose the pointer in the exchange order list. Just update the original allocated type. Maybe you have a better idea. 🤷 

```diff
diff --git a/engine/order_manager.go b/engine/order_manager.go
index 84fb9266a..b4649f2a3 100644
--- a/engine/order_manager.go
+++ b/engine/order_manager.go
@@ -1062,7 +1062,7 @@ func (s *store) getByDetail(det *order.Detail) *order.Detail {
        exchangeOrders := s.Orders[strings.ToLower(det.Exchange)]
        for _, o := range exchangeOrders {
                if o.OrderID == det.OrderID {
-                       return o
+                       *det = *o
                }
        }
        return nil
```

Reply:
You would need to return the det p btw whoopsies.

Reply:
It was intentional, yes.
1) It's more expensive to return copies.
2) For most of my use cases, I want the order to be updated as changes happen to it.
If I don't return a pointer, I'll become incredibly inefficient constantly regrabbing the order all the time.

TL;DR: This actually provides functionality I struggled to live without.

Reply:
I agree with Shazbert. Returning `true` or `false` previously already did the job of seeing if an order exists.

I'm thinking more work would need to be done before we allow the direct order to be returned in the manner that you describe. The order pointer is protected down at that level with mutexes (which don't exist on the order.Detail itself) and also prevents meddling which is why we chose to do all the copying. Having the ability for an order to be modified without knowing the source of the modification is a concern, lowers trust and could lead to some incorrect conclusions being made.

The work done with say the `orderbook.Depth` is a good example of being able to hold an orderbook without being able to harm the underlying linked list and keep getting updates.

Reply:
Understood guys; I'll turn it into a value copy return for now then.

Reply:
@shazbert Strong disagreement about method of gutting and thralling the pointer that was passed in, but I think that's immaterial.
I can achieve what you've asked for 👍

Reply:
Tests will explicitly state behaviour for this